crypto/ake/enclave/src/lib.rs

```diff
@@ -194,10 +230,38 @@ impl<EI: EnclaveIdentity> AkeEnclaveState<EI> {
 
     /// Accept a peer connection
     pub fn peer_accept(&self, req: PeerAuthRequest) -> Result<(PeerAuthResponse, PeerSession)> {
-        self.accept(self.get_peer_self_id()?, &self.peer_inbound, req)
+        let local_identity = self.kex_identity.clone();
+        let ias_report = self.get_ias_report()?;
+
+        // Create the state machine
+        let responder = Start::new(self.get_peer_self_id()?.to_string());
+
+        // Massage the request message into state machine input
+        let auth_request = {
+            let req: Vec<u8> = req.into();
+            NodeAuthRequestInput::<X25519, Aes256Gcm, Sha512>::new(
+                AuthRequestOutput::from(req),
+                local_identity,
+                ias_report,
+                self.get_verifier()?,
+            )
+        };
+
+        // Advance the state machine
+        let mut csprng = McRng::default();
+        let (responder, auth_response) = responder.try_next(&mut csprng, auth_request)?;
+        let session_id = PeerSession::from(responder.binding());
+
+        // This session is established as far as we are concerned.
+        self.peer_inbound.lock()?.put(session_id.clone(), responder);
+
+        // Massage the state machine output into the response message
+        let auth_response: Vec<u8> = auth_response.into();
```

Comment:
What's the reason to do the massaging this way, as opposed to handling it in the PeerAuthResponse::from impl? (Similar above for AuthRequestOutput). I see that this is how it was done before, I'm just curious.

Reply:
This was started this way for dependency reasons: `ClientAuthResponse` and `PeerAuthResponse` couldn't use types from `mc-consensus-api` without pulling in all the gRPC baggage. Doing it in the normal `conversions.rs` way meant `mc-consensus-api`, so bouncing through the `Vec` was the only way to go from one to the other. This code was not changed when the types were refactored into the `mc-attest-enclave-api` crate, and it wasn't changed here either. It's a cleanup thing, but isn't a requirement for launch.

Reply:
👍